src/main/java/io/ebeaninternal/server/deploy/BeanDescriptor.java

```diff
@@ -1993,6 +1992,10 @@ public void sort(List<T> list, String sortByClause) {
   public ElComparator<T> getElComparator(String propNameOrSortBy) {
     ElComparator<T> c = comparatorCache.get(propNameOrSortBy);
     if (c == null) {
+      // This is the same race condition as in DefaultServer and CQueryPlanStats
+      // If avoiding locking is a priority, be sure to investigate Map.computeIfAbsent source code if this is ever refactored to that.
+      // It's a method introduced in java 8 for this exact "if null -> create -> put" scenario.
+      // The conccurrentHashMap implementation has a few synchronization statements in it.
       c = createComparator(propNameOrSortBy);
```

Comment:
It doesn't matter here ... but there are places we can use computeIfAbsent().  

What I would say is doing multiple smaller PR's is **_much_** better for me. More focus, better turn around, less chance of missing something.